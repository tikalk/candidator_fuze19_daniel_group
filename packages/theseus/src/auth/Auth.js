// import auth0 from 'auth0-js'; // no-unused-vars
import autoBind from 'react-autobind';
import history from './history';

export default class Auth {
  // auth0 = new auth0.WebAuth({
  //   domain: process.env.REACT_APP_AUTH0_DOMAIN,
  //   clientID: process.env.REACT_APP_AUTH0_CLIENT_ID,
  //   redirectUri: process.env.REACT_APP_AUTH0_REDIRECT_URI,
  //   audience: process.env.REACT_APP_AUTH0_AUDIENCE,
  //   scope: 'openid email profile read:questions',
  //   responseType: 'token id_token',
  // });

  static getAccessToken() {
    return localStorage.getItem('access_token');
  }

  constructor() {
    autoBind(this);
  }

  login() {
    // this.auth0.authorize();
    setTimeout(() => {
      this.setSession({});
      history.replace('/questions/');
    }, 0);
  }

  handleAuthentication() {
    this.auth0.parseHash((err, authResult) => {
      if (authResult && authResult.accessToken && authResult.idToken) {
        this.setSession(authResult);
        history.replace('/questions/');
      } else if (err) {
        history.replace('/');
        console.log(err); // eslint-disable-line
      }
    });
  }

  setSession(authResult) {
    // Set the time that the access token will expire at
    const expiresAt = JSON.stringify(100000 * 1000 + new Date().getTime());
    localStorage.setItem('access_token', '123');
    localStorage.setItem('id_token', '456');
    localStorage.setItem('expires_at', expiresAt);
    localStorage.setItem('email', 'aa@a.com');
    localStorage.setItem('sub', 'aa');
    localStorage.setItem('name', 'jon dir');
    localStorage.setItem('picture', '');
    localStorage.setItem('hkjsfhlav', authResult);

    // localStorage.setItem('access_token', authResult.accessToken);
    // localStorage.setItem('id_token', authResult.idToken);
    // localStorage.setItem('expires_at', expiresAt);
    // localStorage.setItem('email', authResult.idTokenPayload.email);
    // localStorage.setItem('sub', authResult.idTokenPayload.sub);
    // localStorage.setItem('name', authResult.idTokenPayload.nickname);
    // localStorage.setItem('picture', authResult.idTokenPayload.picture);

    // navigate to the home route
    history.replace('/questions/');
  }

  logout(redirect) {
    // Clear access token and ID token from local storage
    localStorage.removeItem('access_token');
    localStorage.removeItem('id_token');
    localStorage.removeItem('expires_at');
    localStorage.removeItem('email');
    localStorage.removeItem('sub');

    // navigate to the home route
    if (redirect) {
      history.replace('/');
    }
  }

  isAuthenticated() {
    // Check whether the current time is past the
    // access token's expiry time
    const expiresAt = JSON.parse(localStorage.getItem('expires_at'));
    return new Date().getTime() < expiresAt;
  }
}
