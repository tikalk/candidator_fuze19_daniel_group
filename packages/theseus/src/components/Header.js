import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Auth from '../auth/Auth';
import Button from './Button';

const TopBar = styled.div`
  background-color: #3e3e3e;
  padding: 20px;
  color: #e3e3e3;
  display: flex;
  justify-content: space-between;
  font-family: Monaco, Menlo, 'Ubuntu Mono', Consolas, source-code-pro, monospace;
`;

const Title = styled.div`
  font-size: 2rem;
`;

function Header({ auth }) {
  const logout = () => {
    auth.logout(true);
  };

  return (
    <TopBar>
      <Title>Tikal Candidator</Title>
      {auth.isAuthenticated() && <Button onClick={logout}>Log Out</Button>}
    </TopBar>
  );
}

Header.propTypes = {
  auth: PropTypes.instanceOf(Auth).isRequired,
};

export default Header;
