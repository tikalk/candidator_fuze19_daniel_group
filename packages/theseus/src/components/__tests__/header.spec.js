import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';
import Auth from '../../auth/Auth';

jest.mock('../../auth/Auth');

describe('Test <Header>', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('renders correctly - not logged in', () => {
    const auth = new Auth();
    const tree = renderer.create(<Header auth={auth} history={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly - logged in', () => {
    const auth = new Auth();
    auth.isAuthenticated = jest.fn().mockImplementation(() => true);
    const tree = renderer.create(<Header auth={auth} history={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
