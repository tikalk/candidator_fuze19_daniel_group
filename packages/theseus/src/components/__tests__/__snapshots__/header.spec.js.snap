// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Header> renders correctly - logged in 1`] = `
<div
  className="sc-bdVaJa TVzIH"
>
  <div
    className="sc-bwzfXH gllGzP"
  >
    Tikal Candidator
  </div>
  <button
    className="sc-htpNat jZulDA"
    onClick={[Function]}
  >
    Log Out
  </button>
</div>
`;

exports[`Test <Header> renders correctly - not logged in 1`] = `
<div
  className="sc-bdVaJa TVzIH"
>
  <div
    className="sc-bwzfXH gllGzP"
  >
    Tikal Candidator
  </div>
</div>
`;
